js/controller.js

```diff
@@ -1410,24 +1410,45 @@ OO.plugin("Html5Skin", function (OO, _, $, W) {
       if(this.state.isFullScreenSupported) {
         Fullscreen.toggle(this.state.mainVideoInnerWrapper.get(0));
       }
-      // partial support, video element only (iOS)
-      else if (this.state.isVideoFullScreenSupported) {
-        if (this.state.fullscreen) {
-          this.state.mainVideoElement.webkitExitFullscreen();
-        } else {
-          this.state.mainVideoElement.webkitEnterFullscreen();
-        }
+      // partial support, video element only (iOS) and not vr
+      else if (this.state.isVideoFullScreenSupported && !this.videoVr) {
+        this.enterIOSNativeFullscreen();
       } else { // no support
-        if(this.state.isFullWindow) {
-          this.exitFullWindow();
-        } else {
-          this.enterFullWindow();
+        if (this.videoVr) {
+          // if videoVr we don't use native fullscreen (IOS)
+          this.mb.publish(OO.EVENTS.TOGGLE_FULLSCREEN_VR, this.focusedElement);
         }
+        this.enterPseudoFullscreenMode();
       }
+
       this.state.fullscreen = !this.state.fullscreen;
       this.renderSkin();
     },
 
+    /**
+     * Toggle fullscreen for video element only (iOS)
+     * @private
+     */
+    enterIOSNativeFullscreen: function () {
+      if(this.state.fullscreen) {
+        this.state.mainVideoInnerWrapper.removeClass('oo-fullscreen');
```

Comment:
Why are we now removing the class instead of calling `webkitExitFullscreen()`?

Reply:
fixed